src/power_grid_model/validation/errors.py

```diff
@@ -165,18 +167,18 @@ class MultiComponentValidationError(ValidationError):
     E.g. the two fields `id` fields of the `node` and `line` component: [('node', 'id'), ('line', 'id')].
     """
 
-    component: List[str]
-    field: List[Tuple[str, str]]
-    ids: List[Tuple[str, int]]
+    component: List[ComponentType]
+    field: List[Tuple[ComponentType, str]]
+    ids: List[Tuple[ComponentType, int]]
     _message = "Fields {field} are not valid for {n} {objects}."
 
-    def __init__(self, fields: List[Tuple[str, str]], ids: List[Tuple[str, int]]):
+    def __init__(self, fields: List[Tuple[ComponentType, str]], ids: List[Tuple[ComponentType, int]]):
         """
         Args:
             fields: List of field names, formatted as tuples (component, field)
             ids: List of component IDs (not row indices), formatted as tuples (component, id)
         """
-        self.component = sorted(set(component for component, _ in fields))
+        self.component = sorted(set(component for component, _ in fields), key=str)
```

Comment:
wouldn't `component.value` be more correct?

Reply:
@mgovers It really depends. If we add the helper function _str_to_componenttype() to all the error classes' constructors, then yes. As it is now, it can process components both as `str` or `ComponentType`.  Maybe we can discuss a bit about this.